File: languages/go/entity/availability/hosted_publisher.go
package availability

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// HostedPublisher represents a publisher for hosted data availability.
type HostedPublisher struct {
	Type proto.DataAvailabilityType
	Args PublisherArgs
}

// NewHostedPublisher creates a HostedPublisher instance with default publisher arguments.
func NewHostedPublisher() HostedPublisher {
	return HostedPublisher{
		Type: proto.DataAvailabilityType_HOSTED,
		Args: PublisherArgs{},
	}
}

func (e HostedPublisher) ToProto() *proto.Publisher {
	return &proto.Publisher{
		Type: e.Type,
		Args: e.Args.ToProto(),
	}
}

File: languages/js/src/entity/availability/hosted_publisher.ts
import * as proto from "../../bridge/proto/bloock_availability_entities";
import { Publisher } from "./publisher";
import { PublisherArgs } from "./publisher_args";

/**
 * Represents a publisher for hosted data availability.
 */
export class HostedPublisher implements Publisher {
  type: proto.DataAvailabilityType;
  args: proto.PublisherArgs;

  /**
   * Constructs a HostedPublisher object with the specified parameters.
   */
  constructor() {
    this.type = proto.DataAvailabilityType.HOSTED;
    this.args = new PublisherArgs().toProto();
  }

  public toProto(): proto.Publisher {
    return proto.Publisher.fromPartial({
      type: this.type,
      args: this.args
    });
  }
}
